src/redux/colors-by-attribute-slice.tsx

```diff
@@ -0,0 +1,34 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+import { ColorsByAttribute } from "../types";
+import { RootState } from "./store";
+
+// Define a type for the slice state
+interface ColorsByAttributeState {
+  value: ColorsByAttribute | null;
```

Comment:
Could you add TSDoc comments for state items?

Reply:
Done